packages/engine/src/scene/components/ModelComponent.tsx

```diff
@@ -255,7 +256,9 @@ export const useMeshOrModel = (entity: Entity) => {
   const isModel = !!useOptionalComponent(entity, ModelComponent)
   const isChildOfModel = !!useAncestorWithComponent(entity, ModelComponent)
   const hasMesh = !!useOptionalComponent(entity, MeshComponent)
-  return isModel && !isChildOfModel && hasMesh
+  const isPrimitiveGeom = !!useOptionalComponent(entity, PrimitiveGeometryComponent)
+
+  return ((isModel && !isChildOfModel) || isPrimitiveGeom) && hasMesh
```

Comment:
I don't think this logic is right. it'll return false for any entity that doesn't have both a model component and a mesh component on it. The primitive geometry check isn't necessary here either for the same reason as the other comment.

Reply:
i removed the primitive geom check

Reply:
but a model component will always have mesh component

Reply:
I'm not sure that a check for a mesh and model component on the same entity makes a lot of sense, every entity I've tested this with has either a mesh component or a model component and the mesh components are on children entities of the model component.